modzy/models.py

```diff
@@ -29,6 +31,129 @@ def __init__(self, api_client):
+            ForbiddenError: Occurs if the current API client does not have the appropriate entitlements in order
+            to update processing engines
+        """
+        if not max_engines >= min_engines:
+            raise ValueError("Your min_engines value may not exceed the max_engines value")
+
+        model_id = Model._coerce_identifier(model)
+
+        admin_entitlement = "CAN_PATCH_PROCESSING_MODEL_VERSION"
+        data_scientist_entitlement = "CAN_PATCH_MODEL_VERSIONS"
+        admin = self._api_client.accounting.has_entitlement(admin_entitlement)
+        data_scientist = self._api_client.accounting.has_entitlement(data_scientist_entitlement)
+
+        base_request_body = {
+            "minimumParallelCapacity": min_engines,
+            "maximumParallelCapacity": max_engines
+        }
+        base_endpoint = f"{self._base_route}/{model_id}/versions/{version}"
+
+        if admin:
+            endpoint = f"{base_endpoint}/processing"
+            request_body = base_request_body
+        elif data_scientist:
+            endpoint = base_endpoint
+            request_body = {
+                "processing": base_request_body
+            }
+        else:
+            error_message = "You are not authorized to update processing engines. You will need one of the following" \
+                            f"endpoints in order to process: {admin_entitlement}, {data_scientist_entitlement}"
+            raise ForbiddenError(error_message, self._base_route, None)
+
+        try:
+            result = self._api_client.http.patch(endpoint, json_data=request_body)
+            self.logger.info(
+                f"Updated processing engines for Model {model_id} {version}: \n{result['processing']}"
+            )
+        except BadRequestError as e:
+            error_message = e.message
+            self.logger.error(error_message)
+            raise ValueError(error_message)
+
+        if timeout == 0:
+            return
+        else:
+            assert timeout is None or timeout > 0, \
+                "Timeout must either be an integer >= 0 or None to indicate no timeout"
+            start_time = t()
+            while True:
+                current_time = t() - start_time
+                if timeout is not None and current_time > timeout:
+                    self.logger.warning(
+                        f"Timeout of {timeout} seconds reached while waiting for processing engines to initialize."
+                    )
+                    return
+                model_details = self.get_model_details(model_id, version)
+                if model_details is not None:  # This means the model with the id and version is now visible
+                    engines_ready = sum([engine["ready"] for engine in model_details["engines"]])
+                    if engines_ready >= min_engines:
+                        self.logger.info(f"{engines_ready} engines are ready.")
+                        return
+                sleep(poll_rate)
```

Comment:
Busy waits are something that we don't want to do anymore, at least not like the default option, consider that these engine's operations will take an average of 7 minutes (spin-up)? where you will be just hitting the model API service.